lib/controllers/restaurants.js

```diff
@@ -0,0 +1,37 @@
+const { Router } = require('express');
+const authenticate = require('../middleware/authenticate');
+const Restaurant = require('../models/Restaurant');
+const Review = require('../models/Review');
+
+module.exports = Router()
+  .get('/', async (req, res, next) => {
+    try {
+      const restaurant = await Restaurant.getAll();
+      res.json(restaurant);
+    } catch (err) {
+      next(err);
+    }
+  })
+
+  .get('/:id', async (req, res, next) => {
+    try {
+      const restaurant = await Restaurant.getById(req.params.id);
```

Comment:
This would be a good place to use a guard clause. If someone tries to make a get request and provides an id that doesn't exist in the database your `restaurant` variable may not hold the value you are expecting (like `null`), which could cause the next line to fail (because you can't `null.addReviews()`). An example of a guard clause here would be something like:

```js
// if no restaurant object came back move on to the next middleware 
// which is your not-found middleware in app.js
if (!restaurant) next();

Reply:
cool! I actually thought about that line becuase I use it in the get by id ofter, but did not know about using it with get all as I was not certain about the syntax.